evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -1333,30 +1336,30 @@ def get_search_result(
         """
         # we also want to use the default search method
         search_obj = super().get_search_result
-        is_builder = self.locks.check_lockstring(self, "perm(Builder)")
+        is_builder = self.permissions.check("Builder")
+        results = []
 
-        if candidates:
-            candidates = parse_sdescs_and_recogs(
+        if candidates is not None:
+            searched_results = parse_sdescs_and_recogs(
```

Comment:
this step should not modify the candidates, since it's doing an actual search. replacing candidates here broke the ability for builders to search the original candidates by key and instead did the key-search as a global search every time regardless of the candidates.